tests/lib/location/src/location_test.c

```diff
@@ -293,9 +296,13 @@ void test_location_gnss(void)
 		NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START, 0);
 	__wrap_nrf_modem_gnss_start_ExpectAndReturn(0);
 
-	/* TODO: Cannot determine the used system mode but it's set as zero by default in lte_lc */
-	__wrap_nrf_modem_at_scanf_ExpectAndReturn(
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn(
 		"AT%XSYSTEMMODE?", "%%XSYSTEMMODE: %d,%d,%d,%d", 4);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* ltem_mode */
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* nbiot_mode */
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* gps_mode */
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(0); /* mode_preference */
```

Comment:
I would suggest something like this instead. This is in many places, update all if you decide to change these.
```suggestion
	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* LTE-M support */
	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* NB-IoT support */
	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(1); /* GNSS support */
	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_int(0); /* LTE preference */
```